ambry-api/src/main/java/com/github/ambry/clustermap/ReplicaStatusDelegate.java

```diff
@@ -46,7 +46,7 @@ public boolean seal(ReplicaId replicaId) {
   public boolean partialSeal(ReplicaId replicaId) {
     // TODO Efficient_Metadata_Operations_TODO Once the partial seal logic is complete end to end, this should set the
     //  partial seal state.
-    return clusterParticipant.setReplicaSealedState(replicaId, ReplicaSealStatus.NOT_SEALED);
+    return clusterParticipant.setReplicaSealedState(replicaId, ReplicaSealStatus.PARTIALLY_SEALED);
```

Comment:
Before the partial sealed code is fully ready, how do we protect the side effect it may rise after we change the state to PARTIALLY_SEALED?

Reply:
The only side effect will be that some replicas are marked as PARTIALLY_SEALED in helix. But this should not affect anything else. The PutOperation will still pick all the writable partitions only which are the partitions with state `!=PartitionState.READ_ONLY`.